main/test/ca/uwaterloo/flix/language/phase/TestTyper.scala

```diff
@@ -1154,7 +1154,7 @@ class TestTyper extends AnyFunSuite with TestUtils {
         |
       """.stripMargin
     val result = compile(input, Options.TestWithLibMin)
-    expectError[TypeError.ImpureDeclaredAsPure](result)
```

Comment:
Do we need to delete ImpureDeclaredAsPure? If it is no longer used?

Reply:
I was thinking that we could detect if there's an error at the top level and still use this error in that case.

Reply:
(in a future PR)